lib/index.js

```diff
@@ -46,187 +43,103 @@ Ramen.prototype.initialize = function() {
  */
 
 Ramen.prototype.loaded = function() {
-  return is.object(window.Ramen);
+  return !!(window._ramen && window._ramen.push !== Array.prototype.push);
 };
 
-/**
- * Identify.
- *
- * @api public
- * @param {Identify} identify
- */
+Ramen.prototype.group = function(group) {
+  if (!this.identified) {
+    this.identify({ anonymous: true });
+  }
 
-Ramen.prototype.identify = function(identify) {
-  privateGo({ integration: this, identify: identify });
-};
+  var props = group.properties();
+  props = alias(props, { createdAt: 'created' });
+  props = alias(props, { created: 'created_at' });
+  var id = group.groupId();
+  if (id) props.id = id;
 
-Ramen.prototype.group = function(group) {
-  privateGo({ integration: this, group: group });
+  window._ramen.group(props);
 };
 
-Ramen.prototype.page = function(page) {
-  privateGo({ integration: this, page: page });
+Ramen.prototype.page = function() {
+  if (!this.identified) {
+    this.identify({ anonymous: true });
+  }
+
+  window._ramen.page();
 };
 
 Ramen.prototype.track = function(track) {
-  privateGo({ integration: this, track: track });
+  if (!this.identified) {
+    this.identify({ anonymous: true });
+  }
+
+  var properties = track.properties();
+  properties = convertDates(properties, function(date) { return Math.floor(date / 1000); });
+  window._ramen.track(track.event(), properties);
 };
 
-function privateGo(options) {
-  var integration = options.integration;
-  var ramenSettings = {};
-  var property;
-  var identify;
+/**
+ * Identify.
+ *
+ * @api public
+ * @param {Identify} identify
+ */
+
+Ramen.prototype.identify = function(identify) {
+  var user;
   var traits;
   var opts;
-  var user_opts;
-  var created;
-  var name;
-  var id;
-  var email;
-  var group;
-
-  if (options.identify) {
-    identify = options.identify;
-    traits = identify.traits();
-    opts = identify.options(integration.name);
-    user_opts = opts.user;
-    created = identify.created();
-    name = identify.name();
-    id = identify.uid();
-    email = identify.email();
-    group = integration.analytics.group();
-  } else {
-    traits = {};
-    group = {};
-    opts = {};
-  }
 
-  email = email || integration.analytics.user().traits().email;
-  id = id || integration.analytics.user().id();
+  this.identified = true;
 
-  // We need email addresses
-  if (!email) {
+  if (identify.anonymous) {
```

Comment:
Is this in relation to catching what you've been doing in the previous calls where you are calling `this.identify({ anonymous: true });`? Then you'd want this conditional to be `identify.traits().anonymous` -- since `this.identify()` will call `Ramen.prototype.identify`